pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -78,6 +84,101 @@ impl std::fmt::Debug for InMemoryLayerInner {
     }
 }
 
+/// State shared by all in-memory (ephemeral) layers.  Updated infrequently during background ticks in Timeline,
+/// to minimize contention.
+///
+/// This global state is used to implement behaviors that require a global view of the system, e.g.
+/// rolling layers proactively to limit the total amount of dirty data.
+struct GlobalResources {
+    // How many bytes are in all EphemeralFile objects
+    dirty_bytes: AtomicU64,
+    // How many InMemoryLayer objects exist
+    writers: AtomicUsize,
```

Comment:
nit: maybe name this member something else?

Reply:
changed to `dirty_layers`